src/app/components/add-card/add-card.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnDestroy, ViewChild} from '@angular/core';
+import { UtilityService } from 'src/app/services/utility.service';
+import { TaskService } from 'src/app/services/task.service';
+import { Task } from 'src/app/models/Task.model';
+import { NewTaskInputDirective } from 'src/app/directives/new-task-input.directive';
+
+@Component({
+  selector: 'app-add-card',
+  templateUrl: './add-card.component.html',
+  styleUrls: ['./add-card.component.css'],
+})
+
+export class AddCardComponent implements OnDestroy {
+  constructor(
+   private utilityService: UtilityService,
+   private taskService: TaskService
+  ) {}
+
+  @ViewChild(NewTaskInputDirective)
+  newTaskInputDirective!: NewTaskInputDirective;
+  task: Task = new Task();
+  newTask = '';
```

Comment:
```suggestion
  newTask = null;
```

if you need to use string for any reason, add type for this.

Reply:
I need to use string to handle the task input. if user input only spaces as input task then  `this.newTask = this.newTask.replace(/(<([^>]+)>)/g, '').trim();` will trim it and check if task is empty or not.